Update technologies list, reorganize docs (#1182)
CMSgov/managed-care-review: docs/technical-design/testingApproach.md

## Frontend Component Tests

> These tests are written using Jest and Testing Library. We test all our individual components that are big enough to be in Storybook and focus on assertions that confirm that elements render as expect given different sets of props and user interactions.

Reviewer: ```suggestion
These tests are written using Jest and Testing Library. We test all our individual components that are big enough to be in Storybook and focus on assertions that confirm that elements render as expected given different sets of props and user interactions.
```

Example of standard tests for a new form page.

-   Test that for each of the given initial values, the correct values are displayed in fields
-   Test that the form validations run, given some values, the right error renders
-   Test that conditional questions display correctly
    -   E.g. if you select “other” is the “other” field displayed?
-   Test accessibility actions work
    -   E.g. AX focus changes on error

## Cypress Integration Tests

> These end to end tests are written using Cypress and Cypress Testing Library. They are run against a fully deployed version of the app, specifically for test cases that require moving around between different pages, users, and states of the application. While extensive fragile mocking as component tests, should be done here

Reviewer: don't really understand the second sentence here

> These end to end tests are written using Cypress and Cypress Testing Library. They are run against a fully deployed version of the app, specifically for test cases that require moving around between different pages, users, and states of the application. While extensive fragile mocking as component tests, should be done here

Author: ```suggestion
These end to end tests are written using Cypress and Cypress Testing Library. They are run against a fully deployed version of the app, specifically for test cases that require moving around between different pages, users, and states of the application. In addition, tests that would require extensive mocking as Jest tests should be done with Cypress instead.
```
